File: docs/source/ampl/tutorial/ampl_files/sp_lemonade.py
# Load necessary libraries
from amplpy import AMPL
import pandas as pd

# Initialize AMPL object
ampl = AMPL()

# Load the model 
ampl.read("lemonade.mod")

# Set solver option 
ampl.set_option("solver", "highs")

# Solve the problem
ampl.solve()

# Retrieve the results and convert to a Pandas dataframe
solution_df = ampl.get_data("lemonade, iced_tea, profit")
pandas_df = solution_df.to_pandas()

# Perform formatting before printing
pandas_df = pandas_df.T.reset_index() # Transpose and reset indexing
pandas_df.columns = ['item', 'value'] # Provide column names
pandas_df = pandas_df.set_index('item') # Use column 'item' as index

# Print solution
print(pandas_df)

File: docs/source/ampl/tutorial/ampl_files/sp_lemonade.R
# Load necessary libraries
library(rAMPL)

# Initialize AMPL object
ampl <- new(AMPL)

# Load the model
ampl$read("lemonade.mod")

# Set solver option
ampl$setOption("solver", "highs")

# Solve the model
ampl$solve()

# Retrieve the results
solution_df <- ampl$getData("lemonade, iced_tea, profit")

# Perform formatting before printing
solution_df <- as.data.frame(t(solution_df))  # Transpose 
names(solution_df) <- c("value")  # Provide column names

# Print solution
print(solution_df)
